### PR TITLE
Make SSD portable and useable outside the default image & error check when network on a particular node has no services.

### PR DESCRIPTION
**Pull request by dani-docker**
Currently we hardcode the path to the system binaries, example: `/usr/sbin/ipvsadm`  
This PR makes it dynamic by adding a `which` like function in case the binary was in a different path.
The changes make the code portable in case a user wants to run the script by itself or in a different environment/image. 

In addition, a small error check is added in case the network has no services.

Signed-off-by: Dani Louca <dani.louca@docker.com>

### CODE REVIEW

**Comment by dani-docker** [comment]
ping @fcrisciani

File: cmd/ssd/ssd.py [hunk 1]
```diff
@@ -14,6 +14,14 @@
     r'(25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])'
 )
 
+def which(name, defaultPath=""):
+    if defaultPath and os.path.exists(defaultPath):
```
Comment (fcrisciani) [inline cmd/ssd/ssd.py]: Instead of defining it, can we use this: https://docs.python.org/dev/library/shutil.html#shutil.which

Comment (dani-docker) [reply cmd/ssd/ssd.py]: @fcrisciani right, I initially went this route, but `shutil.which` is only available in python 3.0 and has not been back-ported to 2.7, which is the version shipped in the ssd container and in the support dump container.

Comment (fcrisciani) [reply cmd/ssd/ssd.py]: sounds good

File: cmd/ssd/ssd.py [hunk 2]
```diff
@@ -26,13 +34,13 @@ def check_iptables(name, plist):
 
     # get the ingress sandbox's docker_gwbridge network IP.
     # published ports get DNAT'ed to this IP.
-    ip = subprocess.check_output(['/usr/bin/nsenter', '--net=/var/run/docker/netns/ingress_sbox', '/bin/bash', '-c', 'ifconfig eth1 | grep \"inet\\ addr\" | cut -d: -f2 | cut -d\" \" -f1'])
+    ip = subprocess.check_output([ which("nsenter","/usr/bin/nsenter"), '--net=/var/run/docker/netns/ingress_sbox', which("bash", "/bin/bash"), '-c', 'ifconfig eth1 | grep \"inet\\ addr\" | cut -d: -f2 | cut -d\" \" -f1'])
     ip = ip.rstrip()
 
     for p in ports:
-        rule = '/sbin/iptables -t nat -C DOCKER-INGRESS -p tcp --dport {0} -j DNAT --to {1}:{2}'.format(p[1], ip, p[1])
+        rule = which("iptables", "/sbin/iptables") + '-t nat -C DOCKER-INGRESS -p tcp --dport {0} -j DNAT --to {1}:{2}'.format(p[1], ip, p[1])
```
Comment (fcrisciani) [inline cmd/ssd/ssd.py]: just noticed that this is inside a loop so for each ports it is going to be checked, maybe a better approach was to add all the which on top of the file and call them once and maybe error out if some of the tools are missing like `which__or_die`, but you can follow up in a separate patch for it

Comment (dani-docker) [reply cmd/ssd/ssd.py]: Yes, makes sense.
I will push a seperate PR with the suggested changes.
Thx

**Pull request merged**